jsapp/js/account/stripe.api.ts

```diff
@@ -120,17 +119,52 @@ export async function getSubscriptionInterval() {
   return 'month';
 }
 
+const defaultLimits: AccountLimit = Object.freeze({
+  submission_limit: 'unlimited',
+  nlp_seconds_limit: 'unlimited',
+  nlp_character_limit: 'unlimited',
+  storage_bytes_limit: 'unlimited',
+});
+
+function getLimitsForMetadata(
+  metadata: {[key: string]: string},
+  limitsToCompare: false | AccountLimit = false
+) {
+  const limits: Partial<AccountLimit> = {};
```

Comment:
Do we really need this partial here? Is there a way limits can be created without init'ed to {}? Consider most of my comments as optional.

Reply:
We need the limits to represent a subset of `AccountLimit` here because some of the products (like one-time add-ons) include only some of the `AccountLimit` keys.

![image](https://github.com/kobotoolbox/kpi/assets/7819986/41987a81-c4e3-44f4-943e-bbbd6ca410ac)